controllers/admin/AdminThemesController.php

```diff
@@ -789,31 +709,389 @@ private function translateAttributes($attributes)
+            }
+        } catch (Exception $e) {
+            $this->errors[] = $e->getMessage();
+        }
+    }
+
+    /**
+     * Specific postProcess for "deleteTheme" action
+     */
+    protected function postProcessDeleteTheme()
+    {
+        try {
+            $this->validateDeleteAuthorization();
+            $this->theme_manager->uninstall(Tools::getValue('theme_name'));
+            $this->redirect_after = $this->context->link->getAdminLink('AdminThemes');
+        } catch (Exception $e) {
+            $this->errors[] = $e->getMessage();
+        }
+    }
+
+    /**
+     * Specific postProcess for "submitGenerateRTL" action
+     */
+    protected function postProcessSubmitGenerateRTL()
+    {
+        if ((bool)Tools::getValue('PS_GENERATE_RTL')) {
+            Language::installRtlStylesheets(false, true, Tools::getValue('PS_THEMES_LIST'));
+            $this->confirmations[] = $this->trans(
+                'Your RTL stylesheets has been generated successfully',
+                array(),
+                'Admin.Design.Notification'
+            );
+        }
+    }
+
+    /**
+     * Specific postProcess for "resetToDefaults" action
+     */
+    protected function postProcessResetToDefaults()
+    {
+        try {
+            $this->validateAllAuthorizations();
+            if ($this->theme_manager->reset(Tools::getValue('theme_name'))) {
+                $this->confirmations[] = $this->trans(
+                    'Your theme has been correctly reset to its default settings. You may want to regenerate your images. See the Improve > Design > Images Settings screen for the \'Regenerate thumbnails\' button.',
+                    array(),
+                    'Admin.Design.Notification'
+                );
+            }
+        } catch (Exception $e) {
+            $this->errors[] = $e->getMessage();
+        }
+    }
+
+    /**
+     * Specific postProcess for "submitOptionsconfiguration" action
+     */
+    protected function postProcessSubmitOptionsConfiguration()
+    {
+        try {
+            $this->validateAllAuthorizations();
+
+            Configuration::updateValue('PS_IMG_UPDATE_TIME', time());
+
+            if (Tools::getValue('PS_LOGO')) {
+                $this->logo_uploader->updateHeader();
+            }
+            if (Tools::getValue('PS_LOGO_MAIL')) {
+                $this->logo_uploader->updateMail();
+            }
+            if (Tools::getValue('PS_LOGO_INVOICE')) {
+                $this->logo_uploader->updateInvoice();
+            }
+            if (Tools::getValue('PS_FAVICON')) {
+                $this->logo_uploader->updateFavicon();
+                $this->redirect_after = self::$currentIndex . '&token=' . $this->token;
+            }
+
+            Hook::exec('actionAdminThemesControllerUpdate_optionsAfter');
+        } catch (PrestaShopException $e) {
+            $this->errors[] = $e->getMessage();
+            $this->addErrorToRedirectAfter();
+        } catch (Exception $e) {
```

Comment:
$this->errors[] = $e->getMessage(); still needed here, so perhaps remove the } catch (PrestaShopException $e) { and leave the more generic catch

Reply:
I just preserved the original behavior : add an error message if `PrestaShopException`, just change the redirectAfter if generic `Exception`

Reply:
The original behaviour have 

> if (
>                 !in_array(                    $this->authorizationLevel(),
>                      array(AdminController::LEVEL_EDIT, AdminController::LEVEL_ADD, AdminController::LEVEL_DELETE))
>                  || _PS_MODE_DEMO_
>              ) {
>                  $this->errors[] = $this->trans('You do not have permission to edit this.', array(), 'Admin.Notifications.Error');

you're loosing it in you catch (Exception $e) :)